buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -158,8 +190,14 @@ ByteBuf allocate(int size, int maxCapacity) {
     }
 
     private AdaptiveByteBuf allocate(int size, int maxCapacity, Thread currentThread, AdaptiveByteBuf buf) {
-        long threadId = currentThread.getId();
         int sizeBucket = AllocationStatistics.sizeBucket(size); // Compute outside of Magazine lock for better ILP.
+        FastThreadLocal<Magazine> fastThreadLocalMagazines = this.fastThreadLocalMagazines;
+        if (fastThreadLocalMagazines != null && currentThread instanceof FastThreadLocalThread &&
+            (DEFAULT_USE_CACHED_MAGAZINES_FOR_NON_EVENT_LOOP_THREADS || ThreadExecutorMap.currentExecutor() != null)) {
```

Comment:
I'm not very convinced about having `ThreadExecutorMap.currentExecutor()` in the hot path, really...

Reply:
Sent a203ba3 to address this, as usual, in a very ugly way.